byron/semantics/executable-spec/src/Control/State/Transition/Trace/Generator/QuickCheck.hs

```diff
@@ -204,25 +203,25 @@ shrinkTrace
   -> [Trace sts]
 shrinkTrace baseEnv tr =
     interpretSTS @sts @traceGenEnv baseEnv
-    $ Trace.closure env st0 `traverse` shrunk
+    $ Trace.closure env st0 `traverse` shrinkSignals signals
   where
     env = Trace._traceEnv tr
     st0 = Trace._traceInitState tr
     signals = Trace.traceSignals Trace.NewestFirst tr
 
-    shrunk = sortOn length (shrinkSignals signals)
-
     -- Shrink a list of signals such that we preserve the most recent signal in the shrunk lists.
     -- This shrinker
     --   - recursively omits all but the most recent signal
     --   - builds up lists of signals starting with the most recent signal and
-    --     building up a list excluding the first (oldest) signal
+    --     building up to a list excluding the first (oldest) signal
+    --   - explicitly shrinks in order from small to larger lists of signals
+    --     (i.e. ordered by most to least aggressive shrinking)
     shrinkSignals (sn:_last:[]) =
```

Comment:
Consider exporting this, and add a QC property to check that the resulting list is sorted on length. It might be the case now, but this would prevent regressions.

Reply:
And avoid future headaches.

Reply:
👍